_ramblings/2019-01-05-splitting-cats-and-dogs.md

```diff
@@ -0,0 +1,326 @@
+---
+title: Splitting Cats and Dogs
+layout: rambling
+draft: true
+---
+
+Often when you're programming you find the need to split things up a bit. This
+is such a fundamental part of programming that almost every language gives you
+many tools to aid with this. Maybe it's _modules_, _classes_, _impls_, or
+even _packages_. The list goes on... In fact, one of the best forms of
+splitting can be a simple _function_. Even _comments_ can play a role in this
+if you'd like.
+
+There's a number of reasons you'd want to split up some part of a program. You
+might be looking at a class with the need for countless functions. Instead of
```

Comment:
To me splitting code is also about run time complexity, not just readability, an object that is huge and bloated when I am in the debugger its very hard to keep track of, this is not related to the examples you present here, its just pointing out that this might be an over simplification.

Reply:
@rbritom I'm not disagreeing with you, but I wouldn't call what you're talking about "runtime" complexity, maybe it's cognitive complexity, or I might be tempted to just yell at the damn asswipe that designed it that way.

Runtime complexity should be talking about the performance characteristics.